src/demo.py

```diff
@@ -68,8 +73,11 @@ async def _main():
     password = args.password or getpass("Password: ")
 
     async with aiohttp.ClientSession() as session:
-        opower = Opower(session, utility, username, password)
+        opower = Opower(session, utility, username, password, args.mfa_secret)
         await opower.async_login()
+
+        if args.mfa_secret:
+            sleep(30)  # usually can't reuse TOTP codes

```

Comment:
You just can't log in with the same 6 digit TOTP code twice. TOTP codes switch every 30s and we generate a new one from the TOTP secret every time async_login is called. So we'll either have to do some more session checking (to support session reuse across login calls), or assume that we'll have an interval of >30s between calls to login.

Reply:
> or assume that we'll have an interval of >30s between calls to login

The login intervals from the Home Assistant integration are supposed to be every 12h but for some reason in 2023.8 the login seems to be called twice immediately or maybe even in parallel. I haven't been able to reproduce but I got a lot of bug reports from Exelon and PSE users even from people that were using the custom integration via HACS without any issues in previous core releases. Anyway, I think it's important you fix this somehow in coned.py and remove this sleep from demo.py. BTW, what exception does the 2nd login throw if you remove the sleep? If it's InvalidAuth the integration will enter reauth flow which is undesired since it won't retry and will wait for the user to reenter username/password/secret in the config flow.

Reply:
So I actually encountered this on my instance and this is fixed by keeping the DEVICE_ID cookie around. This will ensure that we don't go through the MFA flow on the second login, therefore we won't have an issue with hitting the OTP twice. I guess demo.py also keeps that cookie around so I could remove this from demo too.

> BTW, what exception does the 2nd login throw if you remove the sleep?

The second auth flow (without fixing up cookies like I'm doing now) hits a 500 on ConEd's side.